### PR TITLE
raftstore: remove checksum calculation when init snapshot

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR tries to remove checksum calculation when initializing a V2::Snap. It fixes the CPU high issue happened in the test environment.

PTAL @zhangjinpeng1987 @siddontang

### CODE REVIEW

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -975,9 +984,8 @@ mod v2 {
                                         meta.get_cf()));
                 }
                 if file_exists(&cf_file.path) {
-                    try!(check_file_size_and_checksum(&cf_file.path,
-                                                      meta.get_size(),
-                                                      meta.get_checksum()));
+                    // Check only the file size for `exists()` to work correctly.
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/snap.rs]: why only check file size can work correctly?

Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: Because `exists` replies on the file size to skip any empty(non-existing) file.

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/raftstore/store/snap.rs [hunk 2]
```diff
@@ -736,27 +736,36 @@ mod v2 {
...
+                                expected_checksum));
         }
         Ok(())
     }
 
+    fn check_file_size_and_checksum(path: &PathBuf,
+                                    expected_size: u64,
+                                    expected_checksum: u32)
+                                    -> RaftStoreResult<()> {
+        check_file_size(path, expected_size)
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: do we have the test to check whether size or checksum is incorrect?

**Comment by siddontang** [comment]
Rest LGTM

File: src/raftstore/store/snap.rs (hunk 2 above)
Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: A test case is added.

**Comment by hhkbp2** [comment]
PTAL @siddontang

File: src/raftstore/store/snap.rs [hunk 3]
```diff
@@ -1637,6 +1651,133 @@ mod v2 {
...
+        // Make all the snapshot in the specified dir corrupted to have incorrect checksum.
+        fn corrupt_snapshot_checksum_in<T: Into<PathBuf>>(dir: T) -> Vec<SnapshotMeta> {
+            let dir_path = dir.into();
+            let mut res = Vec::new();
+            let read_dir = fs::read_dir(dir_path).unwrap();
+            for p in read_dir {
+                if p.is_ok() {
+                    let e = p.as_ref().unwrap();
+                    if e.file_name().into_string().unwrap().ends_with(META_FILE_SUFFIX) {
+                        let mut snapshot_meta = SnapshotMeta::new();
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: I prefer to check the snapshot data, like changing the last byte.

Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: The corruption could be arbitrary. It take a lot of effort to cover all of them.
The test case added is to simulate the small corruption of size/checksum, which relates to this change in this PR.

Comment (siddontang) [inline src/raftstore/store/snap.rs]: yes, I know, but here you change the checksum in meta explicitly. I guess in most cases, the checksum may be right but snapshot data may be corrupted. So I think it is better to change a byte in the snapshot data randomly.

Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: A new test case is added for the way of corruption you required in the comment.

**Comment by hhkbp2** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
Please fix the conflict. @hhkbp2

**Comment by zhangjinpeng1987** [comment]
ci failed

**Pull request merged**